lib: avoided dead stores

DeHess opened the pull request:
Fix coding guideline MISRA C:2012 Rule 2.2 in lib:

> There shall be no dead code

This PR is part of the enhancement issue https://github.com/zephyrproject-rtos/zephyr/issues/48002 which port the coding guideline fixes done by BUGSENG on the https://github.com/zephyrproject-rtos/zephyr/tree/v2.7-auditable-branch back to main

The commit in this PR is a subset of the original auditable-branch commit:
https://github.com/zephyrproject-rtos/zephyr/commit/88608b2e789f159fc59eeb32a407511cbc65aff3

Abramo-Bagnara committed:
coding guidelines: comply with MISRA C:2012 Rule 12.2

- explicit with a cast the destination bitwidth of left shift
  ensuring to not break DTS

Signed-off-by: Abramo Bagnara <abramo.bagnara@bugseng.com>

nordicjm committed:
west: runners: Add run once commands and deferred reset

This adds supports for flashing images with sysbuild where there
are multiple images per board to prevent using the same command per
image flash which might cause issues if they are not ran just once
per flash per unique board name. A deferred reset feature is also
introduced that prevents a board (or multiple) from being reset if
multiple images are to be flashed until the final one has been
flashed which prevents issues with e.g. security bits being enabled
that then prevent flashing further images.

These options can be set at a board level (in board.yml) or a SoC
level (in soc.yml), if both are present then the board configuration
will be used instead of the SoC, and regex can be used for matching
of partial names which allows for matching specific SoCs or CPU cores
regardless of the board being used

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

GhMarwen committed:
drivers: counter: mcux: add support for TPM

Add TPM native Zephyr driver. It's mainly inspired from the GPT
counter implementation.

Signed-off-by: Marouen Ghodhbane <marouen.ghodhbane@nxp.com>
Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

xakep-amatop committed:
drivers: sdhc: add support of Renesas MMC driver

Add basic functionality of Renesas SD/MMC driver. It can be used
for both gen3 and gen4 R-car boards, but tested only with H3ULCB,
Salvator XS M3 and Spider boards. This driver working with SDHC
subsystem.

The driver supports regularal reading/writing throught SD/MMC
controller buffer, DMA mode w/o interrupts and timing tuning.

Add gpio5 and sd0 nodes to h3ulcb and salvator xs which are needed
for working with SD cards. The GPIO node is needed for switching
voltage on SD card through gpio regulator driver.

Notes:
    * the driver doesn't support SPI mode;
    * SCC tuning and DMA mode based on IRQs are enabled by default;
    * an address of a data buffer has to be aligned to 128 bytes if it
      is not, driver will use non-DMA mode automatically;
    * Renesas MMC DMAC doesn't support 64-bit DMA addresses, so for
      case when we have 64-bit xref data address we use non-DMA mode;
    * SD/MMC controller supports block size between 512 and 1 with
      a lot of restrictions, more details you can find in code;
    * support of HS400 mode isn't implemented inside driver.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
boards: arm64: rcar: add board-specific configuration for MMC node

Add pin control group for UHS modes for H3ULCB, Salvator XS M3
boards to appropriate dts files. Both 'uhs' and 'default' pins
states have the same properties for eMMC, e.g. 1.8V.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
tests: add h3ulcb and salvator xs m3configs to perf disk tests

Add MMC related cfgs/overlays of arm64 gen3 rcar boards to perf and
susys/sd/mmc tests.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
samples: littlefs: add support for H3ULCB and Salvator XS M3

Add support of R-Car H3ULCB and Salvator XS M3 boards to
littlefs sample.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

jukkar committed:
net: http: service: Allow user to set TLS security tag list

Have separate macros to setup a HTTPS service.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: lib: http_server: Initial HTTP server support

Original code developed as a GSoC 2023 project by Emna Rekik.

Code refactored in order to provide better bisectability
as the origical commits were not bisectable.

The server supports static and dynamic resources, managed by
HTTP_SERVICE/HTTP_RESOURCE macros.

Fixes #59685
Fixes #59686
Fixes #59688
Fixes #59690
Fixes #59670
Fixes #59700
Fixes #59684
Fixes #59693
Fixes #59693
Fixes #59694
Fixes #59699
Fixes #59696
Fixes #59688
Fixes #59690
Fixes #59670
Fixes #59700
Fixes #59685
Fixes #59686
Fixes #59688
Fixes #59691

Signed-off-by: Emna Rekik <emna.rekik007@gmail.com>
Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

jukkar committed:
tests: net: http_server: Add tests for the HTTP server

Tests for HTTP server support.

Signed-off-by: Emna Rekik <emna.rekik007@gmail.com>
Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

jukkar committed:
samples: net: http_server: Add HTTP server sample application

A simple HTTP server sample application.

Signed-off-by: Emna Rekik <emna.rekik007@gmail.com>
Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

ndrs-pst committed:
bluetooth: hci: refactored bluetooth hci packet type indicators

Introduced a unified definition for HCI packet type indicators in
'bluetooth/hci_types.h. This change streamlines the code in
'drivers/bluetooth/hci/', reducing redundancy.
Enhances maintainability and consistency across all HCI drivers.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

TomasGalbickaNXP committed:
samples: ipc: static_vrings add NXP boards support

This commit adds support for these NXP Multi-core boards for
IPC static_vrings sample using OpenAmp lib:
 - lpcxpresso55s69
 - mimxrt1160_evk
 - mimxrt1170_evk
 - mimxrt1170_evkb

Signed-off-by: Tomas Galbicka <tomas.galbicka@nxp.com>

Commit:
doc: board: ip_k66f: Update doc with LLDP DSA example information

The ip_k66f now supports its KSZ8794CNX switch IC via DSA subsystem.

With the samples/net/dsa it is possible to handle LLDP frames on this
board.

Signed-off-by: Lukasz Majewski <lukma@denx.de>

Commit:
boards: esp32: SDHC support for esp_wrover_kit

Configuration files to enable SDHC in esp_wrover_kit

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

Commit:
drivers: esp32: SDHC implementation

Implementation of SDHC driver for ESP32

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

nordicjm committed:
soc: nordic: Add run once and deferred reset configuration

Adds configuration that allows nRF53 and nRF91-based boards to be
flashed through west using sysbuild for multiple images with the
recover or erase options and prevent running those commands for
each image being flash, which would make the device unbootable.
Also defers reset whilst all images for the cores of these SoCs
are flashed.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

dnikodem committed:
intel_adsp: adsp_memory: update cAVS 2.5 memory definitions

This commit updates the device tree and memory header file
for the Intel cAVS 2.5 platform to define the LSBPM and
HSBPM registers.

Changes include:
- Added node definitions for 'lsbpm' and 'hsbpm' in
intel_adsp_cavs25.dtsi and intel_adsp_cavs25_tgph.dtsi
- Updated adsp_memory.h

Signed-off-by: Damian Nikodem <damian.nikodem@intel.com>

JiafeiPan committed:
dts: mimx93: add tpm counter

Add device tree node for TPM counters.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
tests: counter_basic_api: add imx93 evk support

Add i.MX 93 EVK support for counter_basic_api test application.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

nordicjm committed:
soc: nordic: Reformat soc.yml file

Reformats the soc.yml file to have uniform 2-space indentation

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: build: flashing: Add documentation

Adds documentation on the new flashing configuration system

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: release: 3.7: Add SoC/board flashing configuration note

Adds a note that a new flashing configuration system has been
added for boards and socs

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

JiafeiPan committed:
drivers: clock_control: imx_ccm_rev2: clean up header file

Peripheral mask is 0xFF00, so reorder the peripheral definitions to
be consecutive.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
drivers: clock_control: imx_ccm_rev2: add tpm clock

Add TPM clock support on i.MX 93 platform.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

faxe1008 committed:
modules: lvgl: Fix coordinate handling for invert-{x,y} and swap-xy

This patch fixes two issues in the coordinate handling of the
`zephyr,lvgl-pointer-input` compatible:
- If the swap-xy flag is set the coordinates need to be swapped even
  before the sync event is received.
- The coordinates are stored into an additional variable instead of the
  pending_event. This is done to prevent the double inversion for touch
  release events.

Resolves issue #70539.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

nordicjm committed:
scripts: list_hardware: Add check for soc.yml runner validity

Adds checking that qualifiers listed in a soc.yml file are valid
for the socs and cpuclusters defined in that file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

soburi committed:
boards: arduino: uno_r4: Add Arduino connector for Minima and WiFi

Add Arduino connector definitions for Minima and WiFi

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@fujitsu.com>

cdwilson committed:
boards: shields: Add MikroElektronika Weather Click shield

Add a shield definition for the MikroElektronika Weather Click board.

The Weather Click board features the BME280 sensor, a combined humidity,
pressure, and temperature sensor from Bosch Sensortec.

This shield can be used with any board equipped with a mikroBUS socket
that provides a `mikrobus_i2c` devicetree node label.

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cvinayak committed:
Bluetooth: Controller: Add assertion check for use of scan aux context

Add assertion check for use of scan aux context so that
scan aux context is not assigned again while there is
already an existing LLL scheduling in use by the scan
context.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix BT_CTLR_EARLY_ABORT_PREVIOUS_PREPARE

Fix BT_CTLR_EARLY_ABORT_PREVIOUS_PREPARE to setup new
shorter preempt timeout irrespective of whether there
is a previous prepare enqueued in the prepare pipeline.

Relates to commit d573951f0daf ("Bluetooth: Controller:
Revert back early abort of previous prepare").

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Commit:
west.yml: Update for SDHC driver support

hal_espressif update for SDHC driver support

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

gramsay0 committed:
doc: kernel: code-relocation: Remove kernel example and add tips

The Kernel code relocation example does not work, as the kernel
contains early initialization code that executes before code
relocation takes place

Signed-off-by: Grant Ramsay <gramsay@enphaseenergy.com>

lixuzha committed:
sensing: fix doxygen warnings

Add missing doxygen comments in various sensing headers.

Signed-off-by: Zhang Lixu <lixu.zhang@intel.com>

cvinayak committed:
Bluetooth: Controller: Add prepare pipeline assertion checks

Add prepare pipeline assertion checks to detect faults
like stopping preempt timeout without prior start. Also,
check if the preemptor param is correct when preemption
is performed. Check that the prepare pipeline is not
infinitely looping.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

dnikodem committed:
intel_adsp: adsp_memory: update lnl memory definitions

This commit updates the device tree and memory header file
for the Intel LNL 2.0 platform to define the LSBPM and
HSBPM registers.

Changes include:
- Added node definitions for 'lsbpm' and 'hsbpm' in
intel_adsp_ace20_lnl.dtsi
- Updated adsp_memory.h

Signed-off-by: Damian Nikodem <damian.nikodem@intel.com>

dnikodem committed:
intel_adsp: adsp_memory: update mtl memory definitions

This commit updates the device tree and memory header file
for the Intel MTPM 1.5 platform to define the LSBPM and
HSBPM registers.

Changes include:
- Added node definitions for 'lsbpm' and 'hsbpm' in
intel_adsp_ace15_mtpm.dtsi
- Updated adsp_memory.h

Signed-off-by: Damian Nikodem <damian.nikodem@intel.com>

cvinayak committed:
Bluetooth: Controller: Fix short prepare preempt timeout start

Fix short prepare preempt timeout start, such that if there
are any normal prepare before a short prepare in the
pipeline, then find the short prepare and start its preempt
timeout.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cyliangtw committed:
drivers: adc: support Nuvoton numaker m2l31x

Update m2l31x.dtsi for adc support and update adc_numaker.c
to support acquisition time in 0~255 ADC ticks.

Signed-off-by: cyliang tw <cyliang@nuvoton.com>

omkar3141 committed:
Bluetooth: Mesh: Update models metadata API

This commit updates models metadata API to simplify the usage and
removes the metadata pointer in health server model context.

Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>

dalpra-hcd committed:
boards: stm: nucleo_g4xx: added flash partitions to work with mcuboot

Added code partitioning for ST Nucleo G431RB and G474RE boards.
Defined flash partitions including "mcuboot," "image-0," and "image-1".

Signed-off-by: Hudson C. Dalpra <dalpra.hcd@gmail.com>

cvinayak committed:
Bluetooth: Controller: Use reschedule margin as minimum ticks_slot

Use reschedule margin as minimum ticks_slot when ticker with
reschedule expires. This will ensure not too many unreserved
tickers expire in close proximity that can lead of pile up
of CPU processing time and eventually causing timeout
callbacks to be delayed. This mitigate possible LL asserts
due to overhead in start of radio events.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

rugeGerritsen committed:
Bluetooth: Controller: Default stream count based upon app configs

When building ISO applications on a device that has the host
and controller on the same core, the application developer
shouldn't have to set controller specific configurations.

Without this change the samples iso_receive and iso_broadcast
will fail to run on NRF52 series devices as the samples
try to set up two streams but the controller is configured
to support only one.

Controller unit tests that were previously only enabling the
controller specific ISO configurations now also enable the
top-level ISO configurations to ensure that the default
stream count is properly configured.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

npitre committed:
sys: sflist: allow usage of all available flag bits

Room for flag bits is function of the struct sfnode alignment.
Define the flag mask accordingly. This gives 2 available bits on 32-bit
architectures and 3 bits on 64-bit architectures.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

npitre committed:
sys: sflist: replace unative_t with uintptr_t

We already have uintptr_t which purpose is exactly what we need here.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

cyliangtw committed:
tests: drivers: adc_api: support numaker_m2l31ki

Add support for Nuvoton numaker board numaker_m2l31ki.

Signed-off-by: cyliang tw <cyliang@nuvoton.com>

nordic-krch committed:
tests: drivers: spi: spi_slave: Add nrf54h20dk

Add configuration for nrf54h20dk to the test.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

Commit:
include: mdio.h: Make bus functions optional

Make bus_enable and bus_disable optional for drivers
to implement by checking if it is NULL in the api struct.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
drivers: mdio: Remove empty functions

Remove empty functions that were "implementing" the
mdio bus_enable and bus_disable apis.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

yvanderv committed:
west: linkserver: change to flash command line as required by v1.5.xx

There is no need to specify the core with the flash command line. This
is actually rejected with linkserver v1.5.xx and after.

Signed-off-by: Yves Vandervennet <yves.vandervennet@nxp.com>

diegoherranz committed:
dts: stm32f446: add missing adc2 and adc3

The STM32F446 product line includes 3 ADCs, but adc2 and adc3 were
missing the DTS.

Signed-off-by: Diego Herranz <diegoherranz@diegoherranz.com>

Commit:
pm: device_runtime: Check busy status in runtime_enable

We can't enable device runtime power management in a device that is
set busy since it may suspend this device.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

jori-nordic committed:
Bluetooth: Classic: make SMP use L2CAP BR API

Make `l2cap_br_send_cb()` semi-public, and make SMP use it when it talks
over a BR channel.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jori-nordic committed:
Bluetooth: L2CAP: l2cap_send() -> l2cap_send_sig()

This function is only ever used to send on the signalling channel.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

FRASTM committed:
drivers: watchdog stm32 independent wdg when enable at boot

When the CONFIG_WDT_DISABLE_AT_BOOT is not set, the iwdg
should be configured by iwdg_stm32_install_timeout
and started by iwdg_stm32_setup

Signed-off-by: Francois Ramu <francois.ramu@st.com>

fabiobaltieri committed:
pm: device_runtime: add an access function to get the usage counter

Add a pm_device_usage_get() to get the current usage counter.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
shell: device_service: show the usage conter when pm runtime is enabled

Add an extra "usage=%d" to the device status to show the current usage
counter when PM runtime is enabled.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

abonislawski committed:
kernel: timeslicing: add time slice reset in slice per thread api

This will reset time slice in k_thread_time_slice_set()
when slice per thread api is used.

Currently it will reset it only in standard slice_set

Signed-off-by: Adrian Bonislawski <adrian.bonislawski@intel.com>

ycsin committed:
include: sys: util: implement `NUM_VA_ARGS`

This macro count the number of arguments in the variable
arguments list.

Added test for it in `tests/lib/sys_util`.

Signed-off-by: Yong Cong Sin <yongcong.sin@gmail.com>

fgihl committed:
drivers: sensor: Add TI TMP114 temperature sensors

Add device driver for TI TMP114 I2C temperature sensor.
The driver only support basic functionality, i.e. fetch temperature
using default values.

Datasheet:
	https://www.ti.com/lit/ds/symlink/tmp114.pdf

Signed-off-by: Fredrik Gihl <fgihl@hotmail.com>

fabiobaltieri committed:
input: analog_axis: use k_sem for locking

Swap the mutex with a semaphore for locking the calibration data.
Lighter data structure, no need to use a mutex here.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

Commit:
Revert "drivers: serial: uart_mcux_lpuart: fix IRQ-driven API with ASYNC_API"

This reverts commit a80fa5163bb162f12614dc792aa17016c0fa1ac6.

That commit broke the polling API when used with the interrupt driven,
it is completely wrong to disable the receiver from irq_rx_disable,
the uart_basic_api test has been broken on all lpuart platforms because
of this bug.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
drivers: uart_mcux_lpuart: Fix APIs used together

Handle controller instances wanting to use different
API instances, need to change init flow so that they dont
interfere with each other, trick for now is returning
either the interrupt driven or async apis as not enabled
if the other has already been used on that controller.

Previously, there was an issue where enabling CONFIG_UART_ASYNC_API
would overwrite the init even if controller was meant to use
interrupt driven (as kconfig is global to affect all the controllers)

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

maass-hamburg committed:
mgmt: hawkbit: reset action id

Be able to reset the hawkBit action id.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

cyliangtw committed:
drivers: rtc: support for Nuvoton numaker m46x

Add Nuvoton numaker RTC driver including RTC alarm feature.

Signed-off-by: cyliang tw <cyliang@nuvoton.com>

LukaszMrugala committed:
scripts: twister: Remove Handler atavisms

Removed following fields from relevant Handlers,
as they were unused in code:

* Handler's state
* Handler's generator
* BinaryHandler's call_west_flash
* QEMUHandler's results
* QEMUWinHandler's results

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>

butok committed:
test: mcuboot: enable test for rd_rw612_bga

Enable mcuboot test for rd_rw612_bga

Signed-off-by: Andrej Butok <andrey.butok@nxp.com>

fabiobaltieri committed:
shell: add a callback-filter variant of shell_device_lookup

Add a shell_device_filter variant of the shell device lookup helper that
takes a callback as an argument. This allows more complex filtering on
the device than the existing name prefix.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

ycsin committed:
posix: uname: increase version name length

to 70 to prevent overflowing the char array.

Signed-off-by: Yong Cong Sin <yongcong.sin@gmail.com>

CharlesDias committed:
drivers: video: ov2640: add clock rate control property

Add Clock Rate Control property to OV2640 CMOS video sensor

Signed-off-by: Charles Dias <charlesdias.cd@outlook.com>

peter-mitsis committed:
samples: echo_client: Bump kobj thread bitmask size

Bumps the kobj thread bitmask size due to a recent change
in the logic for aborting threads.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

Commit:
boards: esp32: Alias for conflicting node name (sdhc)

Alias created for sdhc0 SPI device to avoid conflict with
SDIO device names.

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

ubieda committed:
doc: tests: i2c_ram: Specify required Hardware Setup

This testsuite requires an external I2C RAM connected to the I2C bus.
Adding details to provide guidance to users interested in running the
suite.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

fabiobaltieri committed:
doc: services: pm: document the "device list" command

Document the "device list" command in the context of device power
management.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

butok committed:
scripts: runner: linkserver: Add flash support for Intel Hex files

Enables flash support for Intel Hex files,
added with LinkServer v1.5.30.

Signed-off-by: Andrej Butok <andrey.butok@nxp.com>

DeHess committed:
coding guidelines: comply with MISRA C:2012 Rule 11.2

avoid convert pointers to incomplete type using the pointer to first item

Signed-off-by: Hess Nathan <nhess@baumer.com>

rlubos committed:
net: lib: http_server: Add HPACK and Huffman code encoder/decoder

Add HTTP/2 helper libraries to encode and decode HPACK encoded headers,
according to RFC7541.

HPACK string encoding requires to support certain set of Huffman codes,
therefore implement Huffman encoder/decoder as well.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

Thalley committed:
Bluetooth: BAP: Unicast server depend on PACS

When in the BAP unicast server role, at least PAC sink
or PAC source shall be set.

In order to fulfill this new requirement, a few other Kconfig
options had to be changed to a `depends on` from `select` to
avoid recursive Kconfig requirements. This change may require
some applications to update their configurations according
to the migration guide.

The change from `select` to `depends on` is ideal anyhow
as that is the recommended way to add dependencies.

This can checked via the combined BT_PACS Kconfig value.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

FRASTM committed:
boards: arm: stm32f756 with all mpu regions when testing in userspace

Disable the  quadspi mpu region of the nucleo_f756zg (like done on
nucleo_f746zg) when testing the samples/userspace/shared_mem
or tests/kernel/mem_protect/userspace
The stm32f7 cortex M7 has 8 MPU regions and the one for quadspi prevents
the testcase to PASS.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

LukaszMrugala committed:
scripts: twister: Remove harness repeat field

Removed Harness's repeat field.

Added to schemas in e722db14ad27186814fbd2cc82a3b0ec51159d2f,
together with the rest of harness_config.
Added to Harness when it was created.

Now present only in one sample -
mec15xxevb_assy6853 power management sample.

Could not find definitive proof whether it has ever been used.

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>

cvinayak committed:
tests: bsim: Bluetooth: Reduce multiple id test execution iterations

Reduce the iterations in multiple id multiple simultaneous
connections test, and tune the buffer counts to catch any
buffer leak related regressions.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

0x6e committed:
drivers: display: sdl: Fix asserts in sdl_display_write

The asserts and validation code were incorrectly mixing pitch and
width.  These incorrect checks were preventing rendering code from
re-using a smaller section of a buffer allocated to fit the full
screen. We expect to be able to update a portion of the display from a
portion of the buffer, in which case pitch must remain the screen
width. Expecting x + pitch to be smaller than the screen width is
incorrect.

Signed-off-by: Nathan Collins <nathan.collins@kdab.com>

peter-mitsis committed:
kernel: Update thread cpu in z_get_next_switch_handle()

Updates z_get_next_switch_handle() to set the new thread's base.cpu
value as it is done in do_swap(). This helps to ensure that the
last CPU on which the thread executed remains current.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

sylvioalves committed:
twister: update overflow section name regex

esp32 platforms can overflow its dram0_0_seg and dram0_1_seg.
So update current dram section to meet both cases.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

maass-hamburg committed:
drivers: flash: esp32: correct return values

The esp_flash*() functions don't return
negative errno codes, so it's return value
should not be used for the flash api functions.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

duda-patryk committed:
arch: posix: Undefine operating system specific macros for native_sim

Compilers predefine system-specific macros which carry information about
compiler, target architecture and operating system. It provides basic
compiler-dependent information like size of types, their maximal and
minimal values, etc. It allows to write common libc headers for multiple
architectures and operating systems.

These macros allow code to always determine what is the target operating
system. This is a problem when compiling code of modules that supports
multiple operating systems (e.g. cryptography libraries).

To avoid confusion we shouldn't leak host operating system macros (e.g.
__linux__, __linux, linux, etc.) when compiling for native_sim board.

Unfortunately, there is no single universal switch that disables all
operating system macros:
- '-undef' removes also architecture-related macros
- '--target' is only available for Clang compiler

This patch uses '-include' option to include file that undefines all
well-known operating system macros.

Run 'gcc -dM -E - < /dev/null | sort' to get full list of predefined
macros.

Signed-off-by: Patryk Duda <patrykd@google.com>

franc0is committed:
ARC: fault: Fix uninitialized memory access

Found via static analysis. In fault path when checking for stack
overflows, if CONFIG_MULTITHREADING is not set, `guard_end` is left
uninitialized and is subsequently used in a comparison.

The solution is to simply return `false` in this configuration as stack
guards are not configured in the first place.

Signed-off-by: François Baldassari <francois@memfault.com>

cvinayak committed:
Bluetooth: Controller: Enable ticker slot window yield for mesh

Enable use of ticker slow window yield feature under mesh
usecase. This is make sure scan window either slides within
is window or skips to next interval so that it does not
lead to advertising event being skipped.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

evgeniy-paltsev committed:
ARC: nsim_nsim_hs_mpuv6: add missing defconfig

After recent HWv2 model there is defconfig file
missing, so we got incorrect platform configuration. Fix that.

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Signed-off-by: Evgeniy Paltsev <PaltsevEvgeniy@gmail.com>

evgeniy-paltsev committed:
ARC: nsim_hs_mpuv6: fix inconsistent DCCM configuration

The DCCM configuration was set inconsistently between nsim and
mdb arguments

Signed-off-by: Evgeniy Paltsev <PaltsevEvgeniy@gmail.com>
Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>

mjchen0 committed:
drivers: regulator: shell: add is_enabled cmd

Reports whether regulator is enabled or disabled

Signed-off-by: Mike J. Chen <mjchen@google.com>

bravl committed:
samples: drivers: add i2c eeprom target sample

Add simple sample for the i2c target driver interface using the
i2c-target-eeprom as a target device.

Signed-off-by: Bram Vlerick <bram.vlerick@openpixelsystems.org>

nandojve committed:
mgmt: updatehub: Fix json arrays

After the changes introduced by #50816 the UpdateHub could not decode
anymore the JSON object. This introduce missing parsing definitions
to allow JSON parser undertood the correct UpdateHub probe object.

Fixes #69297

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

jvermillard committed:
boards: esp32xx: m5stickc_plus WiFi sample support

Add an overlay to correctely configure the M5 Stick plus WiFi.

Signed-off-by: Julien Vermillard <julien@vermillard.com>

fabiobaltieri committed:
pm: shell: filter out non PM enabled devices

Add a device lookup filter to only suggest devices that are PM enabled.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

rettichschnidi committed:
drivers: gpio: cmake: Prevent conflicts

By sorting the lines alphabetically, conflicts can be reduced.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

rettichschnidi committed:
drivers: gpio: kconfig: Prevent conflicts

By sorting the lines alphabetically, conflicts can be reduced.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

nashif committed:
zbus:  doxygen reference to kconfig

Fix references to kconfig variables.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
clock_control: nrf: fix references to kconfig

Fix references to kconfig variables.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
testsuite: ztress: fix doxygen command

Fix doxygen command.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: various doxygen fixes

Fix paramters and syntax of doxygen in various headers.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

cvinayak committed:
tests: bsim: Bluetooth: Fix sim_length for central_hr_peripheral_hr

Fix sim_length value for central_hr_peripheral_hr so that
the simulation does not end while the test is still in
progress.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

JiafeiPan committed:
soc: imx7d: fix CPP application building error

For CPP application, such as samples/cpp/cpp_synchronization/, it will
report the following building errors:

...
zephyrproject/modules/hal/nxp/imx/devices/MCIMX7D/./MCIMX7D_M4.h:5101:51:
error: 'reinterpret_cast<CCM_Type*>(808976384)' is not a constant
expression
...

The error is caused by commit: 72312feeadcdffd6e0f5d5d0d73a75110c7109b7
" arch: arm: cortex_m: Use cmsis api instead of inline asm in arch_irq_*"
This patch will cause kernel.h includes cmsis_core.h which includes soc.h,
so that soc.h will be used by c++ code.

This patch make soc.h can be c++ compatible.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

nashif committed:
doc: fix duplicate lable data_structures

Fix duplication of label.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

aurel32 committed:
drivers: hwinfo: Add device EUI64 ID support

Some devices, mostly with radio support, have an EUI64 ID (unique and
attributed by IEEE), in addition to the standard device ID.

Add support for reading it through the hwinfo API. As the size is always
the same (8 bytes), there is no need to pass the size in the function,
nor return the number of bytes copied.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

aurel32 committed:
drivers: hwinfo: stm32: Add support for reading EUI64 on STM32W*

The STM32WB, STM32WB and STM32WBA families devices include a EUI64 ID
(called "IEEE 64-bit unique device ID register" in the reference manual
that is distinct from the standard device ID. Add support for reading
it, noting that it is stored in little endian format in the registers.

On other families the weak implementation takes over and returns
-ENOSYS.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

aurel32 committed:
drivers: hwinfo: shell: add support for displaying device EUI64 ID

This adds a 'deveui64' command to display the just added EUI64 device
id. Display "Not supported by hardware" on devices that do not have it.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

ubieda committed:
sensor: rtio_default: Add shift to sensor_submit_fallback DBG printout

To facilitate calculation verification between Q-number and value
obtained through fetch/get APIs.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda committed:
sensor: decoder: Use helper API to compute header size

To be consistent with the computed offset on sensor_submit_fallback()
implementation. This prevents falling in misalignment issues when the
number of sensor channels changes.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

fabiobaltieri committed:
tests: shell: add a test for shell_device_filter

Add a test for shell_device_filter and shell_device_lookup.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
tests: shell: add tests for "device list"

Add tests for the "device list" output with the various configuration of
device power management.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

ubieda committed:
sensor: rtio_default: Remove log when device does not contain channel

Since this loops over all listed sensor channels, this makes more
difficult catching other debug log messages that might be of interest
(e.g: Decoding results).

Signed-off-by: Luis Ubieda <luisf@croxel.com>

wtlee1995 committed:
drivers: flash: andes_qspi: fix shadow variable error

Make the variables naming more specific.

Signed-off-by: Wei-Tai Lee <wtlee@andestech.com>

pdgendt committed:
doc: releases: 3.5: Fix COFNIG typo

Fixing a typo for a `Kconfig` option.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

wtlee1995 committed:
drivers: flash: andes_qspi: fix accessing error variable

Fix accessing error variable when using FLASH_ANDES_QSPI_SFDP_RUNTIME.

Signed-off-by: Wei-Tai Lee <wtlee@andestech.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 15.6

- added missing braces

Signed-off-by: Hess Nathan <nhess@baumer.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 20.9

- avoid to use undefined macros in #if expressions

Signed-off-by: Hess Nathan <nhess@baumer.com>

oyvindronningstad committed:
qemu_malta_qemu_malta_be.yaml: Fix 'identifier' typo

qemu_malta//be -> qemu_malta/qemu_malta/be

This caused the board not to work with Twister.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

maass-hamburg committed:
mgmt: hawkbit: samples: use net config settings

use net config setting and `CONFIG_NET_CONFIG_NEED_IPV4`
to get dhcpv4  address.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 20.9

- avoid to use undefined macros in #if expressions

Signed-off-by: Hess Nathan <nhess@baumer.com>

aescolar committed:
ci: coverage: typo fix in platform list

Fix quemu => qemu

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

DeHess committed:
coding guidelines: comply with MISRA Rule 20.9

- avoid to use undefined macros in #if expressions

Signed-off-by: Hess Nathan <nhess@baumer.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 21.15

- made explicit the copied data type

Signed-off-by: Hess Nathan <nhess@baumer.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 2.2

- avoided dead stores

Signed-off-by: Hess Nathan <nhess@baumer.com>

fabiobaltieri committed:
ci: compliance: upgrade to Python 3.11

The behavior of glob() changedin Python 3.11 and runing the old one
means we are missing some changes that trigger a compliance failure when
ran on an updated system. Upgrade to 3.11 so those are caught in CI.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
MAINTAINERS: fix more paths

Fix few more paths that are failing to parse on system running Python >=
3.11 due to a change in behavior of glob.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

bravl committed:
samples: drivers: i2c: add i2c target api sample

Add simple sample which demonstrates how to setup a custom i2c target.

Signed-off-by: Bram Vlerick <bram.vlerick@openpixelsystems.org>

DeHess committed:
coding guidelines: comply with MISRA Rule 2.2

- avoided dead stores

Signed-off-by: Hess Nathan <nhess@baumer.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 2.2

- avoided dead stores

Signed-off-by: Hess Nathan <nhess@baumer.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 2.2

- avoided dead stores

Signed-off-by: Hess Nathan <nhess@baumer.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 2.2

- avoided dead stores

Signed-off-by: Hess Nathan <nhess@baumer.com>

peter-mitsis commented on lib/os/sem.c:
```diff
@@ -123,7 +123,7 @@ int sys_sem_give(struct sys_sem *sem)
 int sys_sem_take(struct sys_sem *sem, k_timeout_t timeout)
 {
-	int ret_value = 0;
+	int ret_value = k_sem_take(&sem->kernel_sem, timeout);
```
It looks like k_sem_take() is getting called twice in a row.

DeHess replied:
You are right, I fixed it.

peter-mitsis committed:
kernel: Drop non-negative from timeout descriptions

Drops "non-negative" from a few stray kernel API timeout descriptions
as a negative timeout is actually valid and represents an absolute
timeout.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

krish2718 committed:
net: ipv6: mld: Fix improper reference drop

In case of successful submission, the reference shouldn't be put down,
this only should done on error cases.

As reference is put down on success, during the buffer unref, no action
is taken due to an uint8 overflow (ref is now 255), so, the buf->frags
isn't cleared properly and the next time the frags is used and when L2
inserts a second frag, the first head frag and next frag are same (due
to buffer re-use) causing an infinite loop in either net_buf_frag_last
or net_pkt_get_len.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

aurel32 committed:
subsys/settings: add a function to save subtree

When running multiple subsystems or applications on a MCU, the usual
strategy is to use different settings subtrees.

The API provides a way to load or commit a subtree, to avoid adding
dependencies between subsystems or applications, but lacks the way to
save a subtree only. Fix that by adding a settings_save_subtree()
function.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

aurel32 committed:
lorawan/nvm: do not call settings_save() after writing individual settings

The lorawan subsystem do not use settings handler, and instead uses
individual calls to settings_save_one(). With this strategy there is no
need to call settings_save() at the end.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

sylvioalves committed:
tests: boards: add esp32s2_devkitc overlay files

As this module supports ADC, this test needs conf/overlay files
to meet its requirements.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

DeHess committed:
coding guidelines: comply with MISRA Rule 2.2

- avoided dead stores

Signed-off-by: Hess Nathan <nhess@baumer.com>

jori-nordic committed:
Bluetooth: tests: fix `buf` tests

Delete tests that did not end up bringing any value.

What ended up happening is busy-work to "make the test pass" without
understanding what's their original purpose.

Worse, the CI change-based testing is broken and doesn't pick them up,
even by PRs modifying the tests themeselves.
See #68008

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

pdgendt committed:
scripts: build: check_init_priorities.py: Add missing raise for error

An error was not raised.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

bravl committed:
MAINTAINERS.yml: add new samples/drivers/i2c

Add samples/drivers/i2c to the maintainers file so the correct code
owners get added in PRs

Signed-off-by: Bram Vlerick <bram.vlerick@openpixelsystems.org>

ycsin committed:
include: sys: util: add note and test for NUM_VA_ARGS_LESS_1

Notes that `NUM_VA_ARGS_LESS_1` support up to 64 arguments,
and added test for it in `tests/lib/sys_util`.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

sylvioalves committed:
west.yaml: update hal_espressif after SDHC porting

Update revision reference for PR #71165 as needed.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

nashif committed:
ci: check_compliance: use inclusive language

UNDEF_KCONFIG_WHITELIST ->   UNDEF_KCONFIG_ALLOWLIS

Fixes #72044

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

DeHess committed:
Merge branch 'misra_rule_2_2_auditable_to_main_lib' of https://github.com/DeHess/zephyr into misra_rule_2_2_auditable_to_main_lib

DeHess committed:
Merge branch 'misra_rule_2_2_auditable_to_main_lib' of https://github.com/DeHess/zephyr into misra_rule_2_2_auditable_to_main_lib

lorc commented:
Could you please rebase this PR onto main branch to limit the impact?
